pilot/pkg/config/coredatamodel/controller.go

```diff
@@ -200,18 +200,27 @@ func (c *Controller) Apply(change *mcpclient.Change) error {
 	}
 	c.configStoreMu.Unlock()
 
+	// only dispach add/update/delete events for service entries
+	if descriptor.Type == model.ServiceEntry.Type {
+		c.dispatchServiceEntryEvents(innerStore, prevStore)
+	}
+
```

Comment:
The ServiceEntry change looks good. I think we'll still need to generate a single dummy event for other configuration to force Pilot to invalidate the cache and pickup the latest set of changes. This should be easy to confirm as most e2e tests should fail if we're not handling this case correctly.